components/ui/serverEventsCategorized/DateFilterBadges.tsx

```
@@ -0,0 +1,76 @@
+import Badge from "@components/ui/common/badge";
+import { buildCanonicalUrl } from "@utils/url-filters";
+import type { CategorySummaryResponseDTO } from "types/api/category";
+
+interface DateFilterBadgesProps {
+  placeSlug: string;
+  categorySlug?: string;
+  categories?: CategorySummaryResponseDTO[];
+  contextName: string;
+  ariaLabel?: string;
+}
```

ESLint

[error] Do not declare interfaces outside of the /types directory. Centralize all interfaces in /types.

(no-restricted-syntax)